sympy/functions/combinatorial/numbers.py

```diff
@@ -2044,3 +2044,139 @@ def nT(n, k=None):
     for discard in m.enum_range(n[_M], k-1, k):
         tot += 1
     return tot
+
+
+#-----------------------------------------------------------------------------#
+#                                                                             #
+#                          Motzkin numbers                                    #
+#                                                                             #
+#-----------------------------------------------------------------------------#
+
+
+class motzkin(Function):
+    """
+    The nth Motzkin number is the number
+    of ways of drawing non-intersecting chords
+    between n points on a circle (not necessarily touching
+    every point by a chord). The Motzkin numbers are named
+    after Theodore Motzkin and have diverse applications
+    in geometry, combinatorics and number theory.
+
+    Motzkin numbers are the integer sequence defined by the
+    initial terms `M_0 = 1`, `M_1 = 1` and the two-term recurrencerelation
+    `M_n = \frac{2*n + 1}{n + 2} * M_{n-1} + \frac{3n - 3}{n + 2} * M_{n-2}`.
+
+
+    Examples
+    ========
+
+    >>> from sympy import motzkin
+
+    >>> is_motzkin(4)
+    False
+    >>> find_motzkin_in_range(2,300)
+    [2, 4, 9, 21, 51, 127]
+    >>> find_motzkin_in_range(2,900)
+    [2, 4, 9, 21, 51, 127, 323, 835]
+    >>> find_motzkin_in_range(2,1000)
+    [2, 4, 9, 21, 51, 127, 323, 835]
+    >>> find_first_n_motzkins(10)
+    [1, 1, 2, 4, 9, 21, 51, 127, 323, 835]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Motzkin_number
+    .. [2] https://mathworld.wolfram.com/MotzkinNumber.html
+
+    """
+
+    @staticmethod
+    def is_motzkin(n):
+        try:
+            n = as_int(n)
+        except:
+            return False
+        if n > 0:
+             if n==1 or n==2:
+                return True
+
+             tn1 = 1
+             tn = 2
+             i = 3
+             while tn<n:
+                 a = ((2*i + 1) * tn + (3*i - 3)* tn1) / (i+2)
+                 i+= 1
+                 tn1 = tn
+                 tn = a
+
+             if tn==n:
+                 return True
+             else:
+                 return False
+
+        else:
+            return False
+
+    @staticmethod
+    def find_motzkin_numbers_in_range(x, y):
```

Comment:
You can use first_n_motzkin numbers  and remove  some numbers from start which are less than x.  This will remove redundancy.